mapper-processor/src/test/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoSelectMethodGeneratorTest.java

```diff
@@ -41,19 +41,19 @@ public void should_fail_with_expected_error(String expectedError, MethodSpec met
   public static Object[][] invalidSignatures() {
     return new Object[][] {
       {
-        "Invalid return type: Select methods must return an Entity-annotated class, or a "
-            + "CompletionStage, CompletableFuture, PagingIterable or "
-            + "CompletionStage/CompletableFuture<MappedAsyncPagingIterable> thereof",
+        "Invalid return type: Select methods must return one of [ENTITY, OPTIONAL_ENTITY, "
+            + "FUTURE_OF_ENTITY, FUTURE_OF_OPTIONAL_ENTITY, PAGING_ITERABLE, "
+            + "FUTURE_OF_ASYNC_PAGING_ITERABLE]",
```

Comment:
This is less user-friendly than the previous message, and also gets ugly for `@Query`.

But it has the merit of being exhaustive, and adapting automatically to extensions. Return types are already well covered in the documentation, so I think this could be fine.